### ISSUE TITLE
Consider introducing an HTML and CSS formatter for the Rustup website

### ISSUE THREAD

**Issue by 0xPoe** [rust-lang/rustup#4091]
### Problem you are trying to solve

As dicsusued in https://github.com/rust-lang/rustup/pull/4080#issuecomment-2481060171 and https://github.com/rust-lang/rustup/pull/4080#issuecomment-2485247199. 

I think it would be a good idea to introduce a formatter for this. It would help us avoid similar situations in the future. I’d recommend using Prettier, but it does require adding a new toolchain, such as npm, pnpm, or yarn, to install it. What are your thoughts on this?

### Solution you'd like

1. Install the prettier
2. Format the project using the default configuration.

**Comment by djc** [comment]
> I think it would be a good idea to introduce a formatter for this. It would help us avoid similar situations in the future. I’d recommend using Prettier, but it does require adding a new toolchain, such as npm, pnpm, or yarn, to install it. What are your thoughts on this?

I think the most important factor is that we can easily deploy something into CI. Beyond that, having plugins for common editors (like VS Code) might be a nice to have. I don't have a good sense of the landscape here, Prettier seems okay to me.

**Comment by rami3l** [comment]
@djc Yup, I agree with prettier as well, since it has a pretty sane default config.

**Comment by rami3l** [comment]
Now I'm more inclined to using `oxfmt` since it is a drop-in replacement for `prettier` with `taplo` as a dependency (so that our TOML files are also covered); I've used it in my own projects and it's been doing pretty well.

One remaining concern that I have is that it's not reading the particular config options that we are using for TOML but is passing the default options instead (https://github.com/oxc-project/oxc/issues/18580), so it's not a drop-in replacement for taplo in our case (yet).

I'd love to take on this. Maybe we can use `oxfmt` for JSON/HTML/CSS/YAML etc. and keep `taplo` for TOML for the moment being?

**Comment by djc** [comment]
I've substantially cooled on Taplo, which IIRC is not getting much maintenance.

A Rust-based formatter for HTML/CSS sounds okay though -- I've used Biome in the past for JS/TS but looks like that can also do HTML/CSS.

**Comment by rami3l** [comment]
> I've substantially cooled on Taplo, which IIRC is not getting much maintenance.

It served us well though! For any VSCode users installing the Rust pack it's still the formatter of choice it seems to me.

> I've used Biome in the past for JS/TS but looks like that can also do HTML/CSS.

Yup, I've used Biome in my previous work but unfortunately TOML support seems to be out of scope for their project.

Besides, I highly value an all-in-one formatting solution for our non-Rust files. Considering the fact that we have fewer non-test TOML files flocking around than before (esp. given that we used to be a workspace), we can probably also use the generic `oxfmt` configuration options for TOML instead of `taplo` and enforce some rules manually during review 🙈

**Comment by djc** [comment]
> > I've substantially cooled on Taplo, which IIRC is not getting much maintenance.
> 
> It served us well though! For any VSCode users installing the Rust pack it's still the formatter of choice it seems to me.

How do you think it has served us well? I've mostly not noticed it.

I removed support for it from rustls CI a while ago because it was causing more false positives in CI than actual issues, it doesn't support even pretty basic configuration, and while there seems to be some ongoing maintenance upstream it seems to be basically coasting.

> > I've used Biome in the past for JS/TS but looks like that can also do HTML/CSS.
> 
> Yup, I've used Biome in my previous work but unfortunately TOML support seems to be out of scope for their project.
> 
> Besides, I highly value an all-in-one formatting solution for our non-Rust files. Considering the fact that we have fewer non-test TOML files flocking around than before (esp. given that we used to be a workspace), we can probably also use the generic `oxfmt` configuration options for TOML instead of `taplo` and enforce some rules manually during review 🙈

I would rather use multiple tools that are especially good at what they do instead of focusing on having a single solution for everything -- I don't see strong advantages of having a single tool for TOML as well as HTML and CSS, since they serve pretty different needs.

That said, I don't have strong opinions on Biome vs oxfmt -- I don't remember why I picked Biome for my project.

**Comment by rami3l** [comment]
> How do you think it has served us well? I've mostly not noticed it.

Maybe it's just me, but I have the impression that previously we had some review comments specifically regarding TOML styling (especially when adjusting dependencies), and the fact that they are gone is, I believe, because a standard formatter and its config are in place.

> I removed support for it from rustls CI a while ago because it was causing more false positives in CI than actual issues, it doesn't support even pretty basic configuration

That's quite interesting, and I'd love to see what those false positives might actually be to make a more informed decision...

In any way I agree that I can focus on HTML/CSS for now and we can discuss about TOML separately 🙏
